src/Control/Comonad/Trans/Env.hs

```diff
@@ -145,6 +149,22 @@ runEnv (EnvT e (Identity a)) = (e, a)
 runEnvT :: EnvT e w a -> (e, w a)
 runEnvT (EnvT e wa) = (e, wa)
 
+#if MIN_VERSION_base(4,9,0)
+instance (Eq e, Eq1 w) => Eq1 (EnvT e w) where
+  liftEq eq (EnvT e1 wa1) (EnvT e2 wa2) = e1 == e2 && liftEq eq wa1 wa2
+
+instance (Ord e, Ord1 w) => Ord1 (EnvT e w) where
+  liftCompare comp (EnvT e1 wa1) (EnvT e2 wa2) = case compare e1 e2 of
+    EQ -> liftCompare comp wa1 wa2
+    r -> r
+
+instance (Show e, Show1 w) => Show1 (EnvT e w) where
```

Comment:
If we're going to define a `Show1` instance, we should probably also add a `Read1` instance, no?

Reply:
Heh, I suppose. I'd rather have neither, but Show is just too ubiquitous to ignore. So yeah, I'll add a Read1 to go with it.

Reply:
How about adding a dependency on deriving-compat, rather than adding a manual `Read1` instance (and also use that to skip the other instances)?

Reply:
Unfortunately, I think the use of Template Haskell will be a non-starter here.